native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/deserializer/avro/AvroDeserializer.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using Avro;
+using Avro.Generic;
+using Avro.IO;
+using Avro.Specific;
+using AWSGsrSerDe.common;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace AWSGsrSerDe.deserializer.avro
+{
+    /// <summary>
+    /// Avro specific de-serializer responsible for handling the Avro protocol
+    /// specific conversion behavior.
+    /// </summary>
+    public class AvroDeserializer : IDataFormatDeserializer
+    {
+        private readonly MemoryCache _datumReaderCache;
+        private AvroRecordType _avroRecordType;
+
+        /// <summary>
+        /// Size of the datum reader cache for testing purpose
+        /// </summary>
+        public int CacheSize => _datumReaderCache.Count;
+        
+        private AvroDeserializer()
+        {
+            // TODO: make the cache size limit configurable
+            _datumReaderCache = new MemoryCache(new MemoryCacheOptions { SizeLimit = 1000 });
+        }
+
+        /// <summary>
+        /// Constructor for AvroDeserializer that accepts configuration elements
+        /// </summary>
+        /// <param name="configs">configuration elements</param>
+        public AvroDeserializer(GlueSchemaRegistryConfiguration configs)
+            : this()
+        {
+            Configure(configs);
+        }
```

Comment:
Is this still needed?

Reply:
we still need to configure avro record types that the de-serializer should expect right?